opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,480 @@
+/*
+ * Copyright (c) 2014-2021 Intel, Inc. All rights reserved.
+ * Copyright (c) 2015      Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2017-2022 Amazon.com, Inc. or its affiliates.
+ *                         All Rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef OPAL_ACCELERATOR_H
+#define OPAL_ACCELERATOR_H
+
+#include "opal/class/opal_object.h"
+#include "opal/mca/mca.h"
+
+BEGIN_C_DECLS
+
+/**
+ * Accelerator flags
+ */
+/* Unified memory buffers */
+#define MCA_ACCELERATOR_FLAGS_UNIFIED_MEMORY 0x00000001
+
+/**
+ * Transfer types.
+ * UNSPEC - Not specified
+ * HTOH - Host to Host
+ * HTOD - Host to Device
+ * DTOH - Device to Host
+ * DTOD - Device to Device
+ */
+typedef enum {
+    MCA_ACCELERATOR_TRANSFER_UNSPEC = 0,
+    MCA_ACCELERATOR_TRANSFER_HTOH,
+    MCA_ACCELERATOR_TRANSFER_HTOD,
+    MCA_ACCELERATOR_TRANSFER_DTOH,
+    MCA_ACCELERATOR_TRANSFER_DTOD,
+} opal_accelerator_transfer_type_t;
+
+struct opal_accelerator_stream_t {
```

Comment:
I don't understand why you created `opal_accelerator_stream_t` and `opal_accelerator_event_t` structures?  Why don't you just pass these as void pointers in the interface?  Why create a structure around a void pointer, what does that get you?

Reply:
This was actually a request from Brian. What I assume this allows is for more flexibility for the implementer (ie, they can add fields to the object as needed)

Reply:
I think I missed this because it was resolved by someone, but I wasn't asking for a named structure with a single void* inside of it.  I was asking that we avoid void* in our APIs, because it's hard to figure out object types in either side of the code, and very easy to feed the wrong type of object.

But I was also asking that streams and events get made into OMPI objects, which means that the user of the API has a nice, opaque object it can use, and each component can have a customized storage for streams and events (that match their implementations) by inheriting from the base stream/event structures.  So you have an abstract object for callers that can be used for data storage by the implementations.  But this requires making these proper OMPI objects, which also changes your API (as mentioned in comments later in this file).

But, to be clear, I see no value in a structure with a single member that is a void*.  If you're going to do that, just typedef opal_accelerator_stream_t to be a void*.